src/express/state.js

```diff
@@ -0,0 +1,29 @@
+export const state = {
+  products: {
```

Comment:
You're working with array-like objects, why ot's worth using it? Better create an array of object with `id` field and afterwards search for desirable object with `Array.find()`

Reply:
Instead of using array of objects, I used Entity pattern. So, we can directly access product with id easily. [link](https://stackoverflow.com/questions/38445006/state-as-array-of-objects-vs-object-keyed-by-id). Please provide your comment on this.

Reply:
Okay, it's good. But what if your key/id will be a UUID or a string? You will access it through `products['myID']`, but assume we have a real DB and it returns the collection (Array) of data, not an object. Yes, sometimes you may use array-like object, but that's not the case

Reply:
Ok. Got it.